From 92c62df07a776760535c48e90e68bff28053b850 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:47 -0500
Subject: [wireless] ath5k: fix hardware queue index assignment

Message-id: <1418207389-7899-34-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101605
O-Subject: [RHEL7.1 PATCH 33/35] ath5k: fix hardware queue index assignment
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

BZ#1155538

Backport from wireless-testing of:

commit 9e4982f6a51a2442f1bb588fee42521b44b4531c
Author: Felix Fietkau <nbd@openwrt.org>
Date:   Sun Nov 30 21:52:57 2014 +0100

    ath5k: fix hardware queue index assignment

    Like with ath9k, ath5k queues also need to be ordered by priority.
    queue_info->tqi_subtype already contains the correct index, so use it
    instead of relying on the order of ath5k_hw_setup_tx_queue calls.

    Cc: stable@vger.kernel.org
    Signed-off-by: Felix Fietkau <nbd@openwrt.org>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Complied only.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/ath/ath5k/qcu.c b/drivers/net/wireless/ath/ath5k/qcu.c
index 0583c69..ddaad71 100644
--- a/drivers/net/wireless/ath/ath5k/qcu.c
+++ b/drivers/net/wireless/ath/ath5k/qcu.c
@@ -225,13 +225,7 @@ ath5k_hw_setup_tx_queue(struct ath5k_hw *ah, enum ath5k_tx_queue queue_type,
  } else {
   switch (queue_type) {
   case AR5K_TX_QUEUE_DATA:
-   for (queue = AR5K_TX_QUEUE_ID_DATA_MIN;
-    ah->ah_txq[queue].tqi_type !=
-    AR5K_TX_QUEUE_INACTIVE; queue++) {
-
-    if (queue > AR5K_TX_QUEUE_ID_DATA_MAX)
-     return -EINVAL;
-   }
+   queue = queue_info->tqi_subtype;
    break;
   case AR5K_TX_QUEUE_UAPSD:
    queue = AR5K_TX_QUEUE_ID_UAPSD;
-- 
1.7.1